server/s3_copy_object_action.cc

```diff
@@ -82,6 +82,158 @@ void S3CopyObjectAction::setup_steps() {
   ACTION_TASK_ADD(S3CopyObjectAction::send_response_to_s3_client, this);
 }
 
+void S3CopyObjectAction::get_source_bucket_and_object() {
```

Comment:
hi amit...we need to talk on this function where it should be,,,because for authorization also we will need this bucket name and object name...but this function here is after authorization.

Reply:
@amitwac2608 : I think, you may have to override 'S3Action::load_metadata() in your new action class to load all metadata about destination bucket and object, and source bucket (all the data that you need for CopyObject operation).

Reply:
I have also added a new function in RequestObject class to read source bucket and object. Authserver can use that